external-crates/move/move-vm/runtime/src/paranoid_type_checker.rs

```diff
@@ -0,0 +1,908 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use move_binary_format::errors::Location;
+use std::sync::Arc;
+
+use move_binary_format::{
+    errors::{PartialVMError, PartialVMResult, VMResult},
+    file_format::{Ability, AbilitySet, Bytecode},
+};
+use move_core_types::{account_address::AccountAddress, vm_status::StatusCode};
+use move_vm_types::{loaded_data::runtime_types::Type, values::Locals};
+
+use crate::{
+    interpreter::{check_ability, FrameInterface, InstrRet, InterpreterInterface},
+    loader::{Function, Loader, Resolver},
+    plugin::Plugin,
+};
+
+const OPERAND_STACK_SIZE_LIMIT: usize = 1024;
```

Comment:
I think that's fair. I just wanted to maintain parity w/ the existing behavior. But if we only care about the operand_stack.values exceeding limit then let's get rid of this

Reply:
it's more that the type stack is 1:1 with the operand stack, when they were together one could claim it would be easier to hold that invariant (though it was not really), but at the same time it was an intrusive concept to have next to the stack. If the 2 stack go out of sync things would be pretty bad, so I am not clear there is much value.
But at the same time we are not using the paranoid checks so it would not matter much.
Again more something to think about for now